[다오] step-1 체스 프로젝트 시작

imojaek opened the pull request:
## 구현 내용

- gradle 프로젝트를 만들고, gradle을 통해 실행해본다.
- JUnit과 assertJ 이용해서 테스트 코드를 작성해본다.

## 고민 사항

-  결국 요구되는 작동에는 문제가 없게 만들었지만, 개인 환경 powershell에서 한글깨짐을 해결해야한다.

## 기타

imojaek committed:
Gradle 첫 설정

Maven이 아닌 Gradle 사용을 위해서 콘솔에서 gradle init으로 프로젝트를 생성했습니다.

imojaek committed:
Pawn 테스트

미리 작성되어 있는 PawnTest의 테스트에 맞춰 Pawn클래스를 생성하고,
생성자, 필드, getColor를 만들었습니다.

imojaek committed:
검은 폰 테스트 추가

Pawn 생성자로 "black" 을 전달해서 검은색 폰인 pawnBlack을 만들었습니다.
그리고 pawnBlack의 color가 "black"인지 확인합니다.

imojaek committed:
PawnTest create() 리팩토링

중복으로 사용되는 white, black 상수화,
색이 제대로 들어가는지 확인하는 verifyPawnColor 작성

imojaek committed:
build.gradle main 위치 수정

App 으로 되어있었는데, chess라는 패키지를 추가함으로써 chess.App 으로 변경했습니다.

honux77 commented on app/src/test/java/chess/AppTest.java:
이 파일은 제거해 주는 게 좋겠네요.

imojaek replied:
사용되지 않는 파일임을 확인하고, 제거했습니다!

honux77 commented on app/src/test/java/chess/PawnTest.java:
```diff
@@ -0,0 +1,22 @@
+    public void create() {
+        final String colorWhite = "white";
+        final String colorBlack = "black";
+        verifyPawnColor(colorWhite);
```
```suggestion
        verifyPawnColor(WHITE);
```
일반 문장처럼 코드에서도 단어의 중복은 피하고 간결하게 작성하는 게 좋습니다.

imojaek replied:
이후 작성에서도 중복되는 의미의 반복은 피하도록 고려하겠습니다

imojaek committed:
PawnTest 상수명의 상수화

colorBlack, colorWhite 이던 상수의 이름을 중복되는 의미의 color를 제거하고 대문자 BLACK, WHITE로 변경했습니다.

imojaek committed:
AppTest 제거

gradle 생성과정에서 만들어져 "gradle test" 의 작동을 확인했던 AppTest.java 파일을 제거했습니다.

honux77 commented on app/src/test/java/chess/PawnTest.java:
```diff
@@ -0,0 +1,22 @@
+        verifyPawnColor(colorBlack);
+    }
+
+    @DisplayName("폰의 color가 잘 들어가는지 확인한다.")
```
```suggestion
    @DisplayName("지정된 색상의 폰이 생성되어야 한다.")
```
정도 어떨까요?

imojaek replied:
훨씬 좋은 문장입니다... ;ㅅ;

honux77 reviewed (approved):
👍
